### PR TITLE
Commit Shopping Cart

### PR DESCRIPTION
**Pull request by Alex7mob1**

### CODE REVIEW

File: src/main/java/com/opencart/driver/DriverRepository.java [hunk 1]
```diff
@@ -18,6 +20,7 @@ public static void downloadWebDriver() {
 
     public static void instanceWebBrowser() {
         webDriver = new ChromeDriver();
+        webDriver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
```
Comment (adomanich) [inline src/main/java/com/opencart/driver/DriverRepository.java]: Don't use implicitWaits - it is bad practice, remove it, and use explicit ones when necessary

File: src/main/java/com/opencart/pages/EmptyShoppingCartPage.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    public WebElement getEmptyPageContinueButton() {
+        return emptyPageContinueButton;
+    }
+
+    public WebElement getVerifyEmptyBlackWindow() {
+        return verifyEmptyBlackWindow;
+    }
+
```
Comment (adomanich) [inline src/main/java/com/opencart/pages/EmptyShoppingCartPage.java]: remove extra space

File: src/main/java/com/opencart/pages/HomePage.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+public class HomePage extends BasePage {
+
+    @FindBy(xpath = "//*[@class='swiper-slide text-center swiper-slide-duplicate swiper-slide-active']")
+    private WebElement verifyHomePage;
+
+    public WebElement getVerifyHomePage() {
+        return verifyHomePage;
+    }
+
```
Comment (adomanich) [inline src/main/java/com/opencart/pages/HomePage.java]: remove extra space

File: src/main/java/com/opencart/pages/HomePage.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.opencart.pages;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class HomePage extends BasePage {
+
+    @FindBy(xpath = "//*[@class='swiper-slide text-center swiper-slide-duplicate swiper-slide-active']")
```
Comment (adomanich) [inline src/main/java/com/opencart/pages/HomePage.java]: try to use xpath contains() function when you have complex class name

File: src/main/java/com/opencart/pages/containers/ShoppingCartProductContainer.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class ShoppingCartProductContainer extends BasePage {
+
+    public ShoppingCartProductContainer() {
+    }
+
+    public WebElement getTotalPrice(String productName) {
+        return driver.findElement(By.xpath(String.format("//*[@class='table table-bordered']//*[text()='%s']/ancestor::tr//*[@class='text-right'][last()]", productName)));
```
Comment (adomanich) [inline src/main/java/com/opencart/pages/containers/ShoppingCartProductContainer.java]: I am not sure if you need this part of xpath here - "*[@class='table table-bordered']//"

File: src/main/java/com/opencart/pages/containers/ShoppingCartProductContainer.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    }
+
```
Comment (adomanich) [inline src/main/java/com/opencart/pages/containers/ShoppingCartProductContainer.java:18]: remove extra space

File: src/test/java/AddToCartTest.java [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        System.out.println(shoppingCartPageBL.verifyPrice(productName));
+        shoppingCartPageBL.clickOnRemoveWithFewProductsButton()
+                .verifyShoppingCartLocation();
+
+    }
+
+
+
+
+
```
Comment (adomanich) [inline src/test/java/AddToCartTest.java]: remove extra spaces

**Review by adomanich — CHANGES_REQUESTED** [review]

File: src/main/java/com/opencart/pages/containers/ShoppingCartProductContainer.java (hunk 5 above)
Comment (Alex7mob1) [reply src/main/java/com/opencart/pages/containers/ShoppingCartProductContainer.java]: If l don't use this part of the XPath it gets to find more than one locator

**Review by adomanich — APPROVED** [review]

**Review by Zavada-Nazarii — APPROVED** [review]

**Pull request merged**